### PR TITLE
Added Street View RPC

### PR DESCRIPTION
**Pull request by gsteinLTU**
Here's my first take on it. ApiConsumer really makes it easy to add things.

screenshot from 2018-04-17 22 03 04

One thing I wanted an opinion on adding was if I should have it limit invalid requests (e.g. images of bad sizes, lat/long that aren't valid, etc). Google handles some of these for us, but others it just doesn't send an image for.

### CODE REVIEW

File: src/server/rpc/procedures/google-streetview/google-streetview.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ * Terms of use: https://developers.google.com/maps/terms
+ * @service
+ */
+'use strict';
+
+var key = process.env.GOOGLE_MAPS_KEY;
+
+const ApiConsumer = require('../utils/api-consumer');
+const GoogleStreetView = new ApiConsumer('google-streetview', 'https://maps.googleapis.com/maps/api/streetview?',{cache: {ttl: 5*60}});
```
[the remark this answers is not in the document]
Comment (gsteinLTU) [reply src/server/rpc/procedures/google-streetview/google-streetview.js]: This is fixed in a3595a7

File: src/server/rpc/procedures/google-streetview/google-streetview.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        /* eslint-enable no-console*/
+    }
+    return !!key;
+};
+
+/**
+ * Get Street View image of a location using coordinates
+ * @param {Number} width Width of image
+ * @param {Number} height Height of image
+ * @param {Number} latitude Latitude coordinate of location
```
[the remark this answers is not in the document]
Comment (gsteinLTU) [reply src/server/rpc/procedures/google-streetview/google-streetview.js]: This is fixed in a3595a7

File: src/server/rpc/procedures/google-streetview/google-streetview.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ * @param {Number} longitude Longitude coordinate of location
+ * @param {Number} fieldofview Field of View of image
+ * @param {Number} heading Heading of view
+ * @param {Number} pitch Pitch of view
+ * @returns {Image} Image of requested location with specified size and orientation
+ */
+GoogleStreetView.getViewFromLatLong = function(width, height, latitude, longitude, fieldofview, heading, pitch) {
```
[the remark this answers is not in the document]
Comment (gsteinLTU) [reply src/server/rpc/procedures/google-streetview/google-streetview.js]: This is fixed in a3595a7

**Comment by gsteinLTU** [comment]
Parameterized types seems like the best way forward. The Pitch and Heading inputs have the same restrictions that lat and long (respectively) have, but the errors for the existing types would be confusing.

**Comment by brollb** [comment]
So I just opened a PR (https://github.com/NetsBlox/NetsBlox/pull/1990) for adding support for parameterized types - most noteably `BoundedNumber` which can be used to enforce the minimum and maximum values for the input (inclusive). As soon as we merge this, you should be able to change the types for automatic enforcing of the input type and range (eg, `BoundedNumber<-90,90>`).

**Comment by gsteinLTU** [comment]
Wow, that was fast! I'll add those in as soon as I can.

File: src/server/rpc/procedures/google-streetview/google-streetview.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    return !!key;
+};
+
+/**
+ * Get Street View image of a location using coordinates
+ * @param {Number} width Width of image
+ * @param {Number} height Height of image
+ * @param {Number} latitude Latitude coordinate of location
+ * @param {Number} longitude Longitude coordinate of location
+ * @param {Number} fieldofview Field of View of image
```
[the remark this answers is not in the document]
Comment (gsteinLTU) [reply src/server/rpc/procedures/google-streetview/google-streetview.js]: In 224a511 it's all the new bounded types, do you want me to put more information in the description still?

**Comment by brollb** [comment]
This looks good and I am going to merge it but wanted to make a couple notes about the failing checks:
- The codeclimate issues are about the number of arguments (which is fine) 
- The travis build is failing bc of the fork not having the environment variables set. As the tests all pass locally (where the env variables are set), it should be good to go

**Pull request merged**